sorc/ncep_post.fd/INITPOST_NETCDF.f

```diff
@@ -3037,18 +3037,33 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
       VarName='ugrd10m'
       call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
       spval,VarName,u10)
+            
+! 10 m v using gfsio
+      VarName='vgrd10m'
+      call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
+      spval,VarName,v10)
+
+! Both v10 & u10 can be derived from f10m and surface wind in FV3 surface initialization files.
+! If we have f10m, and lack u10 and v10, we derive them here:
+      VarName='f10m'
+      call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
+      spval,VarName,f10m)
+      do j=jsta,jend
+         do i=ista,iend
+            if(u10(i,j) == spval .and. v10(i,j) == spval .and. &
+                f10m(i,j) /=spval .and. uh(i,j,lm)/=spval) then
+               u10(i,j) = f10m(i,j) * uh(i,j,LM)
+               v10(i,j) = f10m(i,j) * vh(i,j,LM)
+            endif
+         enddo
+      enddo
 
       do j=jsta,jend
         do i=ista,iend
           u10h(i,j)=u10(i,j)
         end do
       end do
 !     if(debugprint)print*,'sample l',VarName,' = ',1,u10(isa,jsa)
-            
-! 10 m v using gfsio
-      VarName='vgrd10m'
-      call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
-      spval,VarName,v10)
 
       do j=jsta,jend
```

Comment:
@SamuelTrahanNOAA These lines could be removed with v10h assignment relocated to the above loops.